xilem_core/src/deferred.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use core::{fmt::Display, marker::PhantomData};
+
+use alloc::{boxed::Box, sync::Arc};
+
+use crate::{DynMessage, Message, NoElement, View, ViewId, ViewPathTracker};
+
+/// A `Context` for a [`View`](crate::View) implementation which supports
+/// asynchronous message reporting.
+pub trait AsyncCtx: ViewPathTracker {
+    /// Get a [`Proxy`] for this context.
+    // TODO: Maybe store the current path within this Proxy?
+    fn proxy(&mut self) -> Arc<dyn RawProxy>;
```

Comment:
Maybe the `RawProxy` can be the `MessageProxy<M>` (taking that `TODO` into account)? (Haven't looked into that though whether that's feasable.)

So something like:


```suggestion
    fn proxy<M>(&mut self) -> Arc<dyn MessageProxy<M>>;
```

Reply:
I'm not sure I understand what you're suggesting here.

Reply:
I'm also not entirely sure ^^ (or at least whether that is working for either xilem and xilem_web).
I guess what I mean, is that the `RawProxy` takes a strongly typed `message` in `send_message`, without the path, as it is within the implementor (i.e. the current `MessageProxy` struct as a trait)